src/databricks/labs/ucx/assessment/azure.py

```diff
@@ -498,6 +503,8 @@ def _map_storage(self, storage: AzureResource) -> list[StoragePermissionMapping]
                     StoragePermissionMapping(
                         prefix=f"abfss://{container.container}@{container.storage_account}.dfs.core.windows.net/",
                         client_id=role_assignment.principal.client_id,
+                        principal_id=role_assignment.principal.object_id,
```

Comment:
The issue is the `directoryObjects` api does not return the resource id which is used in UC. See an example response below. The resource id is actually one of the items in `alternativeNames`. So I need object_id to fetch the `alternativeNames` and try match them with the resource id in UC.

```json
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#directoryObjects/$entity",
    "@odata.type": "#microsoft.graph.servicePrincipal",
    "@microsoft.graph.tips": "Use $select to choose only the properties your app needs, as this can lead to performance improvements. For example: GET directoryObjects('<guid>')?$select=deletedDateTime",
    "id": "e6c13c39-41ae-46d5-9361-cfa5066e29b9",
    "deletedDateTime": null,
    "accountEnabled": true,
    "alternativeNames": [
        "isExplicit=False",
        "/subscriptions/3f2e4d32-8e8d-46d6-82bc-5bb8d962328b/resourcegroups/pepsi_rg/providers/Microsoft.Databricks/accessConnectors/ag-conn-access-connector"
    ],
    "appDisplayName": null,
    "appDescription": null,
    "appId": "74d98221-94a4-45da-be65-ebe69250e340",
    "applicationTemplateId": null,
    "appOwnerOrganizationId": null,
    "appRoleAssignmentRequired": false,
    "createdDateTime": "2023-08-22T10:51:01Z",
    "description": null,
    "disabledByMicrosoftStatus": null,
    "displayName": "ag-conn-access-connector",
    "homepage": null,
    "loginUrl": null,
    "logoutUrl": null,
    "notes": null,
    "notificationEmailAddresses": [],
    "preferredSingleSignOnMode": null,
    "preferredTokenSigningKeyThumbprint": null,
    "replyUrls": [],
    "servicePrincipalNames": [
        "74d98221-94a4-45da-be65-ebe69250e340",
        "https://identity.azure.net/VjtXYu8/MPezn5Hmx4p3GTmQnxvf3cz2/3MH6meQUeA="
    ],
    "servicePrincipalType": "ManagedIdentity",
    "signInAudience": null,
    "tags": [],
    "tokenEncryptionKeyId": null,
    "info": null,
    "samlSingleSignOnSettings": null,
    "addIns": [],
    "appRoles": [],
    "keyCredentials": [
        {
            "customKeyIdentifier": "CDE40D4DF02E39BDC7DCF61B77D620D60ED16198",
            "displayName": "CN=74d98221-94a4-45da-be65-ebe69250e340",
            "endDateTime": "2024-04-14T09:28:00Z",
            "key": null,
            "keyId": "12a28c38-c149-4e2e-98f8-d3cdc3338223",
            "startDateTime": "2024-01-15T09:28:00Z",
            "type": "AsymmetricX509Cert",
            "usage": "Verify"
        },
        {
            "customKeyIdentifier": "92E575B7BB4C661E65DE483CC784869E754F37AE",
            "displayName": "CN=74d98221-94a4-45da-be65-ebe69250e340",
            "endDateTime": "2024-02-14T07:19:00Z",
            "key": null,
            "keyId": "73ecc81a-39a3-49b6-99e4-3b4df13cb6a0",
            "startDateTime": "2023-11-16T07:19:00Z",
            "type": "AsymmetricX509Cert",
            "usage": "Verify"
        },
        {
            "customKeyIdentifier": "2F7F3E9B72A950227B8F5619062D43ECC9EBDB0A",
            "displayName": "CN=74d98221-94a4-45da-be65-ebe69250e340",
            "endDateTime": "2023-11-20T10:46:00Z",
            "key": null,
            "keyId": "7500be61-6c47-499b-a714-7a9b83ade126",
            "startDateTime": "2023-08-22T10:46:00Z",
            "type": "AsymmetricX509Cert",
            "usage": "Verify"
        }
    ],
    "oauth2PermissionScopes": [],
    "passwordCredentials": [],
    "resourceSpecificApplicationPermissions": [],
    "verifiedPublisher": {
        "displayName": null,
        "verifiedPublisherId": null,
        "addedDateTime": null
    }
}
```

Reply:
we can extract the resource id from the `alternativeNames` and put it in `role_assignment.principal`, so I don't need the object_id and don't need to call `directoryObjects` again in my class. The concern is if this makes `azure_storage_account_info.csv` complex for customer to read?

Reply:
Don't leak internal identifiers!